File: BackEnd/app/Models/Portfolio.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;


class Portfolio extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'title',
        'description',
        'link',
        'photo',
        'files',
        'user_id',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function files()
    {
        return $this->hasMany(PortfolioFile::class);
    }

    public function skills(): BelongsToMany
    {
        return $this->belongsToMany(Skill::class, 'portfolio_skill', 'portfolio_id', 'skill_id')->withTimestamps();
    }
}

File: FrontEnd/Mobile/lib/models/portfolio.dart
import 'package:jobera/models/portfolio_file.dart';
import 'package:jobera/models/skill.dart';

class Portfolio {
  final int id;
  final String title;
  final String description;
  final String link;
  final String? photo;
  final List<PortfolioFile> files;
  final List<Skill> skills;

  Portfolio({
    required this.id,
    required this.title,
    required this.description,
    required this.link,
    required this.skills,
    required this.files,
    this.photo,
  });

  Portfolio.fromJson(Map<String, dynamic> json)
      : id = json['id'] as int,
        title = json['title'] as String,
        description = json['description'] as String,
        link = json['link'] as String,
        photo = json['photo'] as String?,
        files = [
          for (var portofolioFile in json['files'])
            (PortfolioFile.fromJson(portofolioFile)),
        ],
        skills = [
          for (var skill in json['skills']) (Skill.fromJson(skill)),
        ];

  Portfolio.empty()
      : id = 0,
        title = '',
        description = '',
        link = '',
        photo = '',
        files = [],
        skills = [];
}
